File: test-suite/src/main/java/example/micronaut/MusicianAssistant.java
package example.micronaut;

import dev.langchain4j.data.message.ChatMessage;
import dev.langchain4j.data.message.SystemMessage;
import dev.langchain4j.data.message.UserMessage;
import dev.langchain4j.model.chat.ChatModel;
import dev.langchain4j.model.chat.response.ChatResponse;
import jakarta.inject.Singleton;

import java.util.List;

@Singleton
public class MusicianAssistant {
    private static final SystemMessage SYSTEM_MSG = SystemMessage.from("""
      You are an expert in Jazz music.
      Reply with only the names of the artists, albums, etc.
      Be very concise.
      If a list is given, separate the items with commas.""");

    private final ChatModel model;

    public MusicianAssistant(ChatModel model) { // <1>
        this.model = model;
    }

    public Musician generateTopThreeAlbums(String name) {
        List<ChatMessage> messages = generateTopThreeAlbumsMessages(name);
        ChatResponse albums = model.chat(messages);
        String topThreeAlbums = albums.aiMessage().text();
        return new Musician(name, topThreeAlbums);
    }

    private static List<ChatMessage> generateTopThreeAlbumsMessages(String name) {
        return List.of(SYSTEM_MSG, UserMessage.from(
            String.format("Only list the top 3 albums of %s", name)
        ));
    }
}

File: test-suite-python/src/main/python/example/micronaut/MusicianAssistant.py
from dev.langchain4j.data.message import SystemMessage, UserMessage
from dev.langchain4j.model.chat import ChatModel
from jakarta.inject import Singleton
from java.util import List

from .Musician import Musician

SYSTEM_MSG = SystemMessage.from_("""\
      You are an expert in Jazz music.
      Reply with only the names of the artists, albums, etc.
      Be very concise.
      If a list is given, separate the items with commas.""")


@Singleton
class MusicianAssistant:

    def __init__(self, model: ChatModel):  # <1>
        self.model = model

    def generate_top_three_albums(self, name: str) -> Musician:
        messages = generate_top_three_albums_messages(name)
        albums = self.model.chat(messages)
        top_three_albums = albums.aiMessage().text()
        return Musician(name, top_three_albums)


def generate_top_three_albums_messages(name: str):
    return List.of(SYSTEM_MSG, UserMessage.from_(
        f"Only list the top 3 albums of {name}"
    ))
